tools/system_libs.py

```diff
@@ -978,18 +981,21 @@ def get_cflags(self):
       cflags += ['-DUSE_WEBGL2=1', '-s', 'USE_WEBGL2=1']
     if self.is_ofb:
       cflags += ['-D__EMSCRIPTEN_OFFSCREEN_FRAMEBUFFER__']
+    if self.is_full_es3:
+      cflags += ['-D__EMSCRIPTEN_FULL_ES3__']
     return cflags
 
   @classmethod
   def vary_on(cls):
-    return super(libgl, cls).vary_on() + ['is_legacy', 'is_webgl2', 'is_ofb']
+    return super(libgl, cls).vary_on() + ['is_legacy', 'is_webgl2', 'is_ofb', 'is_full_es3']
```

Comment:
Do we really want to introduce another variant?  Why not just always include these and let metadce remove them when possible?

Reply:
The problem is GetProcAddress. If an app uses that, metadce can't remove GL functions (as it maps string names to them, the user can ask for any one). And as a result we end up with the entirety of libgl linked in. And then it's a link error as the JS library side intentionally does not link in these functions (so that GetProcAddress will return "null" if asked for them, which is the correct behavior if FULL_ES3 is not used).

Reply:
Right so any app that GLGetProcAddress implicitly ends up with `FULL_ES3` .. is that such a big deal?

Reply:
Yes, an app not declaring FULL_ES3 has a guarantee of better performance. By default we don't support non-WebGL2-friendly methods, it's opt-in.

Reply:
(I don't know if it's a big deal, but that is the design we've had for this.)

Reply:
OK lgtm.  I guess there is a tradeoff we need to make somewhere between our internal implementation costs and complexity and micro optimizations like this.